src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -651,8 +627,15 @@ class RaftIvfIndexNode : public IndexNode {
 
 void
 SetRaftMemPool(size_t init_size, size_t max_size) {
-    LOG_KNOWHERE_INFO_ << "Set GPU pool size: init size " << init_size << ", max size " << max_size;
+    // TODO(wphicks): Is this actually used anywhere? If so, when in the
```

Comment:
Was not sure if this is being used at all or if we exclusively set the pool size by querying the environment variable.

If we actually use this somewhere, we should ensure that it happens before the `gpu_resources` object is initialized and then use the stored values for configuring the pool allocator on creation.

Reply:
This api is used for init when milvus starts.

Reply:
milvus can parse the yaml configuration to set this option.

Reply:
Aha! Perfect. In that case, we can bring this back in. Any issue with letting it accept `std::optional<std::size_t>` instead of `size_t` to allow us to defer to the default rmm configuration if no option is provided?